crates/matrix-sdk-ui/src/timeline/mod.rs

```diff
@@ -309,76 +341,120 @@ impl Timeline {
+        let mention_the_sender = if self.room().own_user_id() == replied_to_info.sender {
             AddMentions::No
         } else {
             AddMentions::Yes
         };
 
-        let content = match reply_item.content() {
-            TimelineItemContent::Message(msg) => {
+        let content = match replied_to_info.content {
+            ReplyContent::Message(msg) => {
                 let event = OriginalRoomMessageEvent {
                     event_id: event_id.to_owned(),
-                    sender: reply_item.sender().to_owned(),
-                    origin_server_ts: reply_item.timestamp(),
+                    sender: replied_to_info.sender,
+                    origin_server_ts: replied_to_info.timestamp,
                     room_id: self.room().room_id().to_owned(),
                     content: msg.to_content(),
                     unsigned: Default::default(),
                 };
                 content.make_reply_to(&event, forward_thread, mention_the_sender)
             }
-            _ => {
-                let Some(raw_event) = reply_item.latest_json() else {
-                    return Err(UnsupportedReplyItem::MISSING_JSON.into());
-                };
-
-                content.make_reply_to_raw(
-                    raw_event,
-                    event_id.to_owned(),
-                    self.room().room_id(),
-                    forward_thread,
-                    mention_the_sender,
-                )
-            }
+            ReplyContent::Raw(raw_event) => content.make_reply_to_raw(
+                &raw_event,
+                event_id.to_owned(),
+                self.room().room_id(),
+                forward_thread,
+                mention_the_sender,
+            ),
         };
 
         self.send(content.into()).await?;
 
         Ok(())
     }
 
+    /// Gives the information needed to reply to an event from an event id.
+    pub async fn get_replied_to_info_from_event_id(
+        &self,
+        event_id: &EventId,
+    ) -> Result<RepliedToInfo, UnsupportedReplyItem> {
+        if let Some(timeline_item) = self.item_by_event_id(event_id).await {
+            return timeline_item.get_replied_to_info();
+        }
+
+        let event = match self.room().event(event_id).await {
+            Ok(event) => event,
+            Err(error) => {
+                error!("Failed to fetch event with ID {event_id} with error: {error}");
+                return Err(UnsupportedReplyItem::MISSING_EVENT);
+            }
+        };
+
+        // We need to synthetise the TimelineitemContent and we can do that by casting
+        // the event as a `AnySyncTimelineEvent` which is the same as a
+        // `AnyTimelineEvent`, but without the `room_id` field.
+        // The cast is valid because we are just losing track of such field.
+        let raw_sync_event: Raw<AnySyncTimelineEvent> = event.event.cast();
+        let Ok(sync_event) = raw_sync_event.deserialize_as::<AnySyncTimelineEvent>() else {
```

Comment:
nit: instead of using `deserialize_as` with the explicit generic type, you can use `deserialize()` here.